diff --git a/BUILD b/BUILD
index d667123b24..81009b48e0 100644
--- a/BUILD
+++ b/BUILD
@@ -4823,6 +4823,9 @@ grpc_cc_library(
         "grpc++_public_hdrs",
         "grpc_public_hdrs",
     ],
+    external_deps = [
+        "absl/status"
+    ],
 )
 
 grpc_cc_library(
diff --git a/bazel/grpc_extra_deps.bzl b/bazel/grpc_extra_deps.bzl
index a370fe4496..106a16ec65 100644
--- a/bazel/grpc_extra_deps.bzl
+++ b/bazel/grpc_extra_deps.bzl
@@ -51,10 +51,6 @@ def grpc_extra_deps(ignore_version_differences = False):
 
     api_dependencies()
 
-    go_rules_dependencies()
-    go_register_toolchains(version = "1.20")
-    gazelle_dependencies()
-
     # Pull-in the go 3rd party dependencies for protoc_gen_validate, which is
     # needed for building C++ xDS protos
     go_third_party()
diff --git a/include/grpcpp/impl/status.h b/include/grpcpp/impl/status.h
index 94c0e80a9b..755a136e24 100644
--- a/include/grpcpp/impl/status.h
+++ b/include/grpcpp/impl/status.h
@@ -21,6 +21,7 @@
 
 // IWYU pragma: private, include <grpcpp/support/status.h>
 
+#include "absl/status/status.h"
 #include <grpc/status.h>
 #include <grpc/support/port_platform.h>
 #include <grpcpp/support/config.h>
@@ -107,6 +108,10 @@ class GRPC_MUST_USE_RESULT_WHEN_USE_STRICT_WARNING GRPCXX_DLL Status {
         error_message_(error_message),
         binary_error_details_(error_details) {}
 
+  operator absl::Status() const& {
+    return absl::Status(static_cast<absl::StatusCode>(code_), error_message_);
+  }
+
   // Pre-defined special status objects.
   /// An OK pre-defined instance.
   static const Status& OK;
